### ISSUE TITLE
input seg length

### ISSUE THREAD

**Issue by xinqipony** [tatp22/linformer-pytorch#2]
great work!
I noticed the linformer input is (batch_size, seq_len, channels), can seq_len be variable length or should the attention be masked if seq_len is padded? why seq_len is a fixed length?

**Comment by tatp22** [comment]
Hi! The reason why `seq_len` is variable length is because the way I implemented it, the learned `E` and `F` matrices are [`nn.Linear` layers](https://github.com/tatp22/linformer-pytorch/blob/7872830f16d35979e91bb67c88aa65d3675d24f1/linformer_pytorch/linformer_pytorch.py#L12), which have to know their sequence length beforehand. To that end, however, I created the `Padder` class, which automatically pads your input sequence if it happens to be of a length other than `input_size`. To use it in your code, use it like [here](https://github.com/tatp22/linformer-pytorch#padder)
